fsspec/implementations/cached.py

```diff
@@ -232,6 +233,41 @@ def clear_cache(self):
         rmtree(self.storage[-1])
         self.load_cache()
 
+
+    def clear_expired_cache(self, expiry_time=None):
+        """Remove all expired files and metadata from the cache
+        Parameters
+        ----------
+        expiry_time: int
+            The time in seconds after which a local copy is considered useless.
+            If not defined the default is equivalent to the attribute from the
+            file caching instantiation.
+        In the case of multiple cache locations, this clears only the last one,
+        which is assumed to be the read/write one.
+        """
+        if not expiry_time:
+            expiry_time = self.expiry
+
+        self._check_cache()
+
+        for path, detail in self.cached_files[-1].copy().items():
+            if time.time() - detail["time"] > expiry_time:
+                if self.same_names:
+                    basename = os.path.basename(detail["original"])
+                    fn = os.path.join(self.storage[-1], basename)
+                else:
+                    fn = os.path.join(self.storage[-1], detail["fn"])
+                if os.path.exists(fn):
+                    os.remove(fn)
+                    self.cached_files[-1].pop(path)
+
+        if self.cached_files[-1]:
```

Comment:
What is this bit doing?

Reply:
Not really sure what you are referring to as multiple parts are included but I'll try to explain both. 

lines 261/262 are the core component that removes the local file and the corresponding entity in the cache list.

line  264 checks if, for any reason, there are no left files and triggers a complete cleaning.